Library/Formula/azure-cli.rb

```diff
@@ -0,0 +1,38 @@
+class AzureCli < Formula
+  desc "Official Azure CLI"
+  homepage "http://aka.ms/dhf67j"
+  url "https://github.com/Azure/azure-xplat-cli/archive/v0.9.12-November2015.tar.gz"
+  version "0.9.12"
+  sha256 "7b92d0965deb01bc27a7790e36d171b5e2287f25145ad084238b6dcc48beec79"
+
+  # install from the dev branch to get the latest
+  head "https://github.com/Azure/azure-xplat-cli.git", :branch => "dev"
+
+  option "without-completion", "Azure CLI bash completion will not be installed"
+
+  depends_on "node"
+
+  def install
+    ENV.prepend_path "PATH", "#{Formula["node"].opt_libexec}/npm/bin"
+
+    # install node dependencies
+    system "npm", "install"
+    # remove windows stuff
+    rm_rf "bin/windows"
+    # only install the minimal stuff
+    (prefix/"src").install Dir["lib", "node_modules", "package.json", "bin"]
+    # symlink azure
+    bin.install_symlink (prefix/"src/bin/azure")
+
+    ENV.prepend_path "PATH", bin
+    if build.with? "completion"
+      # add completion if it's requested
+      system "azure --completion >> #{prefix}/azure.completion.sh"
```

Comment:
Probably prefer reading with backticks and writing with `Pathname.write` here, thanks.